cmd/go-vcs/go-vcs.go

```diff
@@ -309,23 +310,38 @@ func main() {
 		}
 
 	case "branches":
-		if len(args) != 0 {
-			log.Fatal("branches takes no arguments.")
+		if len(args) > 1 {
+			log.Fatal("branches takes 0 or 1 arguments.")
+		}
+
+		// Open using go/vcs to figure out VCS type (git, hg).
+		r := vcs2.New(".")
```

Comment:
I just added it for testing, I didn't mean to keep it. I needed it because the repo I was testing with was "git", not "hg".

As far as extracting it into here, I can't extract just that function, since you'd need the entire `github.com/shurcooL/go/vcs` library. It's a relatively small library, but it's more than just one func. See for yourself:

https://sourcegraph.com/github.com/shurcooL/go@master/.tree/vcs

However, what I can (and want) to do is extract the entire `go/vcs` Go package into a standalone git repository. Would that be better?

Eventually, I'd like to merge the functionality of `shurcooL/go/vcs` together with `sourcegraph/go-vcs`, or rather make them more compatible and play better together. Right now it's a little awkward because they share some responsibilities but differ in other aspects.